flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java

```diff
@@ -312,22 +304,23 @@ void releaseBuffer() {
 
 		@Override
 		ByteBuf write(ByteBufAllocator allocator) throws IOException {
-			checkNotNull(buffer, "No buffer instance to serialize.");
```

Comment:
shouldn't we keep `checkState(buffer instanceof NetworkBuffer)`? Otherwise isn't that an untested and unused code path? (I'm not sure, so I will relay on your judgement)

Reply:
actually, we don't rely on any `NetworkBuffer` specifics (except for the need to set an allocator in that case)